source/core/slang-io.cpp

```diff
@@ -125,6 +132,32 @@ namespace Slang
 #endif
 	}
 
+    /* static */SlangResult Path::GetCanonical(const String & path, String & canonicalPathOut)
+    {
+#if defined(_WIN32)
+        // https://msdn.microsoft.com/en-us/library/506720ff.aspx
+        wchar_t* absPath = ::_wfullpath(nullptr, path.ToWString(), 0);
+        if (!absPath)
+        {
+            return SLANG_FAIL;
+        }  
+
+        canonicalPathOut =  String::FromWString(absPath);
+        ::free(absPath);
+        return SLANG_OK;
+#else
+        // http://man7.org/linux/man-pages/man3/realpath.3.html
+        char* canonicalPath = ::realpath(path.begin(), nullptr);
```

Comment:
Eventually we should double-check if this function is available on all *nixes, or if we need to add a third path to cover MacOS (I'm figuring Windows+Linux+MacOS covers the main platforms we'll care about for the immediate future).

Reply:
It's fine on OSX (I just tested it).
Looks like BSD does https://www.unix.com/man-page/freebsd/3/realpath/